.github/workflows/lint-build.yml

```
@@ -0,0 +1,58 @@
+name: Lint & Build
+
+on:
+  workflow_dispatch:
+  pull_request:
+  push:
+    branches: [main]
+
+env:
+  HAS_USERSTYLES_TOKEN: ${{ secrets.USERSTYLES_TOKEN != '' }}
+
+jobs:
+  lint:
+    name: Lint Userstyles
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Setup Deno
+        uses: nekowinston/setup-deno@main
+        with:
+          deno-version: v1.x
+
+      - name: Run Lint
+        run: deno task lint
+
+  build:
+    name: Generate Stylus Export
+    runs-on: ubuntu-latest
+    needs: lint
```

We only want to build on our repo, right?
```suggestion
    needs: lint
    if: env.HAS_USERSTYLES_TOKEN
```
we can remove other `if` checks as well.

---

What about the final step, should we not still check if were on the main branch, or shall we move that to earlier on?

---

Just checking `github.ref == 'refs/heads/main'` should be fine I think.

---

After thinking about this again, checking github.repository && github.ref is the best way, since a fork could get this to run, when an env variable is added for some reason.
The clone will fail if the token isn't present, so we don't need to check that here AFAIK.